wiki/Client/File_formats/Osu_(file_format)/en.md

```diff
@@ -4,346 +4,623 @@
+
+### Hit circle
+
+#### Parameters
+
+`x,y,start_time,object_type,hit_sound,hit_sample`
+
+| Option | Value type | Description | Value |
+| :-- | :-- | :-- | :-- |
+| `x` | Integer | The location of the hit circle on the X axis in pixels | `96` |
+| `y` | Integer | The location of the hit circle on the Y axis in pixels | `52` |
+| `start_time` | Integer | Time in milliseconds when the hit circle is to be hit | `1279` |
+| `object_type` | Integer | Bit flags indicating the type of object | bit `0` = `1` constant |
+| `hit_sound` | Integer | Bit flags indicating the hit sound applied to the object | `0` |
+| `hit_sample` | Colon-separated list | Information about which samples are played when the object is hit | `0:0:0:0` by default |
 
-#### Slider curves
+#### Examples
 
-When constructing curves for a slider, `x` and `y` are used for the first point, and `curvePoints` supply the rest.
+`104,59,35660,1,0,0:0:0:0:`
 
-There are four types of slider curves in osu!:
+`96,52,1279,5,0,0:0:0:0:`
 
-- **Bézier (B):** [Bézier curves](https://en.wikipedia.org/wiki/Bézier_curve) of arbitrary degree can be made. Multiple bézier curves can be joined into a single slider by repeating their points of intersection.
-- **Centripetal catmull-rom (C):** [Catmull curves](https://en.wikipedia.org/wiki/Centripetal_Catmull–Rom_spline) are an interpolating alternative to bézier curves. They are rarely used today due to their lack of visual appeal.
-- **Linear (L):** These curves form a straight path between all of their points.
-- **Perfect circle (P):** Perfect circle curves are limited to three points (including the hit object's position) that define the boundary of a circle. Using more than three points will result in the curve type being switched to bézier.
+### Slider
 
-If the slider's `length` is longer than the defined curve, the slider will extend until it reaches the target length:
+#### Parameters
 
-- For bézier, catmull, and linear curves, it continues in a straight line from the end of the curve.
-- For perfect circle curves, it continues the circular arc.
+`x,y,start_time,object_type,hit_sound,curve_type|curve_points,slides,length,edge_sounds,edge_sets,hit_sample`
 
-*Notice: The slider's `length` can be used to determine the time it takes to complete the slider. `length / (SliderMultiplier*100)*beatLength` tells how many milliseconds it takes to complete one slide of the slider (assuming `beatLength` has been adjusted for inherited timing points).*
+| Option | Value type | Description | Value |
+| :-- | :-- | :-- | :-- |
+| `x` | Integer | The location of the first slider point on the X axis in pixels | `96` |
+| `y` | Integer | The location of the first slider point on the Y axis in pixels | `52` |
+| `start_time` | Integer | Time in milliseconds when the slider is to be hit | `12600` |
+| `object_type` | Integer | Bit flags indicating the type of object | bit `1` = `1` constant |
```

Comment:
`bit 1 = 1 constant` is a bit confusing IMO. Is it supposed to be `bit 1 = 2 constant`?